src/block/block_compact.c

```diff
@@ -75,18 +75,20 @@ __wt_block_compact_end(WT_SESSION_IMPL *session, WT_BLOCK *block)
  *     Collect compact progress stats.
  */
 void
-__wt_block_compact_get_progress_stats(WT_SESSION_IMPL *session, WT_BM *bm,
-  uint64_t *pages_reviewedp, uint64_t *pages_skippedp, uint64_t *pages_rewrittenp,
-  uint64_t *pages_rewritten_expectedp)
+__wt_block_compact_get_progress_stats(
+  WT_SESSION_IMPL *session, WT_BM *bm, uint64_t *pages_reviewedp)
 {
     WT_BLOCK *block;
 
     WT_UNUSED(session);
     block = bm->block;
     *pages_reviewedp = block->compact_pages_reviewed;
```

Comment:
Should we get rid of this one too and use `bm->block->compact_pages_reviewed` instead whenever we need to access it?

Reply:
It can be a violation of API usage if it is used directly outside the block code. I feel like keeping the return variable.

Reply:
We access it directly [here](https://github.com/wiredtiger/wiredtiger/blob/b387b765f6f3cbf7f839266a8675a035220c6a4e/src/btree/bt_compact.c#L390-L391) and [here](https://github.com/wiredtiger/wiredtiger/blob/b387b765f6f3cbf7f839266a8675a035220c6a4e/src/btree/bt_compact.c#L340-L342). Are these valid usages?

Reply:
Thanks for finding it, Sean. We are not strictly following the API guidelines. If possible, try to fix them also.

Reply:
I'd prefer a different ticket to fix the existing bugs as this ticket is dedicated to the addition of a new flag.